jax/_src/lax/lax.py

```diff
@@ -2966,6 +2974,25 @@ def _convert_elt_type_pp_rule(eqn, context, settings):
     del params['sharding']  # don't show trivial case
   return core._pp_eqn(eqn.replace(params=params), context, settings)
 
+def _convert_element_type_edtype_rule(ctx, operand, *, new_dtype, weak_type,
+                                     sharding):
+  aval_in, = ctx.avals_in
+  if (aval_in.dtype != new_dtype and
+      ((dtypes.issubdtype(aval_in.dtype, dtypes.extended) and
+        not aval_in.dtype._rules.convert_from(aval_in.dtype, new_dtype)) or
```

Comment:
Can you avoid using `convert_from` and `convert_to`? See https://github.com/jax-ml/jax/pull/23823

Reply:
Sorry this was old code that predated the change. I think we can delete this rule entirely since it should not handle edtypes at all.